odk/src/main/java/org/motechproject/odk/service/impl/VerificationServiceImpl.java

```
@@ -0,0 +1,81 @@
+    private HttpClient client;
+    private RequestConfig config;
+
+    @Autowired
+    public VerificationServiceImpl(HttpClientBuilderFactory httpClientBuilderFactory) {
+        this.config = RequestConfig.custom()
+                .setConnectionRequestTimeout(TIMEOUT)
+                .setConnectTimeout(TIMEOUT)
+                .setSocketTimeout(TIMEOUT)
+                .build();
+        this.client = httpClientBuilderFactory.newBuilder().build();
+
+    }
+
+    @Override
+    public Verification verifyKobo(Configuration configuration) {
+        HttpGet request = new HttpGet(configuration.getUrl() + "/" + configuration.getUsername() + FORM_LIST);
+        request.addHeader(generateBasicAuthHeader(request, configuration));
+        return new Verification(executeRequest(request));
+    }
+
+    @Override
+    public Verification verifyOna(Configuration configuration) {
+        return verifyKobo(configuration);
+    }
+
+    @Override
+    public Verification verifyOdk(Configuration configuration) {
+        HttpGet request = new HttpGet(configuration.getUrl() + FORM_LIST);
+        return new Verification(executeRequest(request));
+    }
+
+    private boolean executeRequest(HttpGet request) {
+        request.setConfig(config);
+        try {
+            HttpResponse response = client.execute(request);
+            return response.getStatusLine().getStatusCode() == STATUS_OK;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    private Header generateBasicAuthHeader(HttpUriRequest request, Configuration configuration) {
+        Header basicAuthHeader;
+        BasicScheme basicScheme = new BasicScheme();
+        try {
+            basicAuthHeader = basicScheme.authenticate(
+                    new UsernamePasswordCredentials(configuration.getUsername(), configuration.getPassword()),
+                    request,
+                    HttpClientContext.create());
+        } catch (Exception e) {
+            return null;
```

Don't like this exception swallow. At best it causes a confusing NPE in the HTTP client code. At worst, authentication is ignored, and you won't figure out why without debuging. Just rethrow this as a runtime exception